homeassistant/components/canary/sensor.py

```diff
@@ -116,6 +115,17 @@ def __init__(
             canary_sensor_type = SensorType.BATTERY
 
         self._canary_type = canary_sensor_type
+        self._attr_state = self.reading
+        self._attr_unique_id = f"{self._device_id}_{self._sensor_type[0]}"
+        self._attr_device_info = {
+            "identifiers": {(DOMAIN, str(self._device_id))},
+            "name": self._device_name,
```

Comment:
Line 99: `self._device_name` can be removed.

```suggestion
            "name": device.name,
```

Reply:
This comment was not fully resolved.